airbyte-commons-worker/src/main/java/io/airbyte/workers/sync/LauncherWorker.java

```diff
@@ -176,13 +176,19 @@ public OUTPUT run(final INPUT input, final Path jobRoot) throws WorkerException
           log.info("Creating " + podName + " for attempt number: " + jobRunConfig.getAttemptId());
           killRunningPodsForConnection();
 
+          // custom connectors run in an isolated node pool from airbyte-supported connectors
+          // to reduce the blast radius of any problems with custom connector code.
+          final var nodeSelectors =
+              isCustomConnector ? workerConfigs.getWorkerIsolatedKubeNodeSelectors().orElse(workerConfigs.getworkerKubeNodeSelectors())
```

Comment:
Similar comment here, I think it's more dangerous to fall back on our primary node selectors, and perhaps we should throw an exception instead.

Reply:
I added this check in `WorkerConfigurationBeanFactory.java` instead - because 1) workerConfigs does not carry `useIsolatedPool` flag, 2) should throw this error as early as possible

Reply:
Nice!